datafusion/physical-expr/src/expressions/in_list.rs

```diff
@@ -300,6 +306,130 @@ fn cast_static_filter_to_set(list: &[Arc<dyn PhysicalExpr>]) -> HashSet<ScalarVa
     }))
 }
 
+fn make_list_contains_decimal(
+    array: &DecimalArray,
+    list: Vec<ColumnarValue>,
+    negated: bool,
+) -> BooleanArray {
+    let contains_null = list
+        .iter()
+        .any(|v| matches!(v, ColumnarValue::Scalar(s) if s.is_null()));
+    let values = list
+        .iter()
+        .flat_map(|v| match v {
+            ColumnarValue::Scalar(s) => match s {
+                Decimal128(v128op, _, _) => v128op.map(|v128| v128),
+                _ => {
+                    unreachable!(
+                        "InList can't reach other data type for decimal data type."
+                    )
+                }
+            },
+            ColumnarValue::Array(_) => {
+                unimplemented!("InList does not yet support nested columns.")
+            }
+        })
+        .collect::<Vec<_>>();
+
+    if !negated {
+        // In
+        array
+            .iter()
+            .map(|v| match v {
+                None => None,
+                Some(v128) => {
+                    if values.contains(&v128) {
+                        Some(true)
+                    } else {
+                        Some(false)
+                    }
+                }
+            })
+            .collect::<BooleanArray>()
```

Comment:
```suggestion
        array
            .iter()
            .map(|v| v.map(|v128| values.contains(&v128)))
            .collect::<BooleanArray>()
```
Is a bit less verbose and perhaps easier to follow

Reply:
The same could be applied to the `NOT IN` branch